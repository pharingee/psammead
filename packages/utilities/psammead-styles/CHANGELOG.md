# @bbc/psammead-styles Changelog

<!-- prettier-ignore -->
| Version | Description |
|---------|-------------|
| 4.0.1 | [PR#2273](https://github.com/bbc/psammead/pull/2273) Snapshot update |
| 4.0.0 | [PR#2229](https://github.com/bbc/psammead/pull/2229) Convert fonts to overridable functions |
| 3.0.0 | [PR#2250](https://github.com/bbc/psammead/pull/2250) Update Mundo, Brasil and Turkish fonts from Helmet to Reith |
| 2.3.0 | [PR#2120](https://github.com/bbc/psammead/pull/2120) Add Scottish colour and scotland service to fonts |
| 2.2.0 | [PR#2021](https://github.com/bbc/psammead/pull/2021) Export global styles |
| 2.1.4 | [PR#1926](https://github.com/bbc/psammead/pull/1926) Update component storybook to use latest inputProvider changes |
| 2.1.3 | [PR#1948](https://github.com/bbc/psammead/pull/1948) Show different font types in storybook |
| 2.1.2 | [PR#1847](https://github.com/bbc/psammead/pull/1847) Fixed Telugu typos |
| 2.1.1 | [PR#1803](https://github.com/bbc/psammead/pull/1803/) Patches broken links on badges in documentation |
| 2.1.0 | [PR#1794](https://github.com/bbc/psammead/pull/1794) Add david dependency badges |
| 2.0.4 | [PR#1777](https://github.com/bbc/psammead/pull/1777) Make psammead-styles treeshakable. |
| 2.0.3 | [PR#1727](https://github.com/bbc/psammead/pull/1727) Fix for Latha Font URL. |
| 2.0.2 | [PR#1682](https://github.com/bbc/psammead/pull/1682) Move all dev dependencies to top level package.json |
| 2.0.1 | [PR#1597](https://github.com/bbc/psammead/pull/1597) Bump @bbc/gel-foundations, @bbc/psammead-storybook-helpers, @bbc/psammead-test-helpers |
| 2.0.0 | [PR#1516](https://github.com/bbc/psammead/pull/1516) Add font-family and font-face declarations for World Service fonts. Changes `getSansItalic` to `getSansRegularItalic`. |
| 1.2.0 | [PR#1234](https://github.com/bbc/psammead/pull/1234) Add custom CSS Grid feature detector |
| 1.1.5 | [PR#1215](https://github.com/bbc/psammead/pull/1215) Fix `sans serif` typo |
| 1.1.4 | [PR#1066](https://github.com/bbc/psammead/pull/1066) Fix Serif Medium font. |
| 1.1.3 | [PR#1180](https://github.com/bbc/psammead/pull/1180) Update devDependencies |
| 1.1.2 | [PR#783](https://github.com/bbc/psammead/pull/783) Update to latest psammead-test-helpers. Update snapshots. |
| 1.1.1 | [PR#769](https://github.com/bbc/psammead/pull/769) Fix stories not appearing in storybook when using `install:packages:link` |
| 1.1.0   | [PR#764](https://github.com/bbc/psammead/pull/764) Add function to get serif bold font |
| 1.0.1   | [PR#677](https://github.com/bbc/psammead/pull/677) Use `@bbc/gel-foundations@3.0.0` |
| 1.0.0   | [PR#679](https://github.com/bbc/psammead/pull/679) Bump version number |
| 0.6.0   | [PR#676](https://github.com/bbc/psammead/pull/676) Add Nassim Persian fonts v1.511 |
| 0.5.0   | [PR#573](https://github.com/bbc/psammead/pull/573) Add functions to get font styles for different services |
| 0.4.1   | [PR#547](https://github.com/bbc/psammead/pull/547) Create story for gel-foundations typographies |
| 0.4.0   | [PR#548](https://github.com/bbc/psammead/pull/548) Bump the reith font version from 2.302 to 2.511 |
| 0.3.5   | [PR#537](https://github.com/bbc/psammead/pull/537) Create story for colours |
| 0.3.4   | [PR#505](https://github.com/bbc/psammead/pull/505) Add Metal; update Pebble |
| 0.3.3   | [PR#424](https://github.com/bbc/psammead/pull/424) Add Snyk badge to readme |
| 0.3.2   | [PR#323](https://github.com/BBC/psammead/pull/323) Update readme storybook badge |
| 0.3.1   | [PR#264](https://github.com/BBC/psammead/pull/319) Resolving package-lock issues. |
| 0.3.0   |  [PR#304](https://github.com/BBC-News/psammead/pull/304) Move font fallback definitions to gel-foundations, add font-face constants. |
| 0.2.1   |  [PR#300](https://github.com/BBC-News/psammead/pull/300) Adds colours for Consent Banner |
| 0.2.0   |  [PR#283](https://github.com/BBC-News/psammead/pull/283) Adds colours and fonts necessary for Simorgh V1.0 :art: |
| 0.1.6   | [PR#224](https://github.com/BBC-News/psammead/pull/224) Add tests for the exported values, coverage 100% :tada: |
| 0.1.5   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.4   | [PR#173](https://github.com/BBC-News/psammead/pull/173) Update PRs welcome link |
| 0.1.3 | [PR#74](https://github.com/BBC-News/psammead/pull/74) Move file contents into a src directory and babel transpile prepublish. |
| 0.1.2 | [PR#72](https://github.com/BBC-News/psammead/pull/72) Update Readme's usage examples |
| 0.1.1 | [PR#67](https://github.com/BBC-News/psammead/pull/67) Update psammead-styles' DOD |
| 0.1.0 | [PR#44](https://github.com/BBC-News/psammead/pull/44) Initial creation of package |
