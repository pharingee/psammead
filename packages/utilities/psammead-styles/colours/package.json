{
  "name": "@bbc/psammead-styles/colours",
  "private": true,
  "main": "../dist/colours",
  "module": "../esm/colours",
  "sideEffects": false
}
