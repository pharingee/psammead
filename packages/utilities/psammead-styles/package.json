{
  "name": "@bbc/psammead-styles",
  "version": "4.0.1",
  "description": "A collection of string constants for use in CSS, containing non-GEL styling details that are bespoke to specific BBC services and products.",
  "repository": {
    "type": "git",
    "url": "https://github.com/bbc/psammead/tree/latest/packages/utilities/psammead-styles"
  },
  "author": {
    "name": "BBC Psammead Maintainers",
    "email": "PsammeadMaintainers@bbc.co.uk"
  },
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/bbc/psammead/issues"
  },
  "homepage": "https://github.com/bbc/psammead/blob/latest/packages/utilities/psammead-styles/README.md",
  "keywords": [
    "bbc",
    "utilities",
    "styles",
    "colours",
    "colors",
    "fonts"
  ]
}
