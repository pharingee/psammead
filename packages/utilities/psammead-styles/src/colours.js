// BBC News Article colours
export const C_EBON = '#222222';
export const C_POSTBOX = '#B80000';
export const C_STORM = '#404040';
export const C_WHITE = '#FFFFFF';
export const C_BLUEJAY = '#0F556C';
export const C_BLUEJAY_LHT = '#C3DEE7';
export const C_OAT_LHT = '#F5F3F1';
export const C_PEBBLE = '#AEAEB5';
export const C_RHINO = '#5A5A5A';
export const C_STONE = '#D5D0CD';
export const C_CHALK = '#ECEAE7';
export const C_SHADOW = '#3F3F42';
export const C_CLOUD_DARK = '#757575';
export const C_CLOUD_LIGHT = '#BABABA';
export const C_LUNAR = '#F2F2F2';
export const C_GHOST = '#FDFDFD';
export const C_METAL = '#6E6E73';

// Colours from other BBC services
export const C_CONSENT_BACKGROUND = '#323232';
export const C_CONSENT_ACTION = '#F6A21D';
export const C_CONSENT_CONTENT = '#BEBEBE';
export const C_ORBIT_GREY = '#4C4C4C';
export const C_DARK_SALTIRE = '#23104C';
