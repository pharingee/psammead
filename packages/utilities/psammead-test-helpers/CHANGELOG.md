# Psammead Test Helpers Changelog

<!-- prettier-ignore -->
| Version | Description |
|---------|-------------|
| 3.0.2 | [PR#2166](https://github.com/bbc/psammead/pull/2166) remove async keyword from test helper to fix regenerator runtime plugin error |
| 3.0.1 | [PR#2123](https://github.com/bbc/psammead/pull/2125) Removes use of async/await |
| 3.0.0 | [PR#2104](https://github.com/bbc/psammead/pull/2104) add function to detect and render helmet for use in snapshots |
| 2.0.2 | [PR#1994](https://github.com/bbc/psammead/pull/1994) Add comment to clarify new shouldMatchSnapshot logic |
| 2.0.1 | [PR#1958](https://github.com/bbc/psammead/pull/1958) Update `shouldMatchSnapshot` to add firstChild logic |
| 2.0.0 | [PR#1917](https://github.com/bbc/psammead/pull/1917) Remove react-test-renderer and shallow rendering |
| 1.3.2 | [PR#1919](https://github.com/bbc/psammead/pull/1919) Accidental version bump |
| 1.3.1 | [PR#1849](https://github.com/bbc/psammead/pull/1849) Update README.md |
| 1.3.0 | [PR#1846](https://github.com/bbc/psammead/pull/1846/) Import test helper functions from simorgh |
| 1.2.1 | [PR#1803](https://github.com/bbc/psammead/pull/1803/) Patches broken links on badges in documentation |
| 1.2.0 | [PR#1794](https://github.com/bbc/psammead/pull/1794) Add david dependency badges |
| 1.1.1 | [PR#1682](https://github.com/bbc/psammead/pull/1682) Move all dev dependencies to top level package.json |
| 1.1.0   | [PR#1233](https://github.com/bbc/psammead/pull/1233) Add ESM modules entry |
| 1.0.2   | [PR#1083](https://github.com/bbc/psammead/pull/1083) Use `jest-styled-components@6.3.3` |
| 1.0.1   | [PR#732](https://github.com/bbc/psammead/pull/732) Use @testing-library/react for generating snapshots |
| 1.0.0   | [PR#679](https://github.com/bbc/psammead/pull/679) Bump version number |
| 0.3.4   | [PR#424](https://github.com/bbc/psammead/pull/424) Add Snyk badge to readme |
| 0.3.3   | [PR#323](https://github.com/BBC/psammead/pull/323) Update readme storybook badge |
| 0.3.2   | [PR#264](https://github.com/BBC/psammead/pull/319) Resolving package-lock issues. |
| 0.3.1   | [PR#264](https://github.com/BBC/psammead/pull/264) Resolving package-lock issues. |
| 0.3.0   | [PR#224](https://github.com/BBC-News/psammead/pull/224) Adds a helper method for testing package export values and uses it to test its own exports. |
| 0.2.1   | [PR#255](https://github.com/BBC-News/psammead/pull/255) Bumps package version to fix what appears to be an erroneously-published version. |
| 0.1.2   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.1   | [PR#173](https://github.com/BBC-News/psammead/pull/173) Update PRs welcome link |
| 0.1.0   | [PR#80](https://github.com/BBC-News/psammead/pull/80) Create initial package, pulled in from [simorgh](https://github.com/BBC-News/simorgh). |
