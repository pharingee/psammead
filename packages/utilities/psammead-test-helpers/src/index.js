import { render } from '@testing-library/react';
import 'jest-styled-components';
import deepClone from 'ramda/src/clone';
import renderWithHelmet from './renderWithHelmet';

export const shouldMatchSnapshot = (title, component) => {
  it(title, done => {
    // select the first child to remove the pointless wrapping div from snapshots
    const removeWrappingDiv = container => container.firstChild;
    renderWithHelmet(component).then(({ container }) => {
      const hasOneChild = container.children.length === 1;
      /*
       * if the container has more than one child then it's a component that uses a
       * fragment at the top level so we should not select the first child because it
       * wouldn't snapshot the whole component
       */
      expect(
        hasOneChild ? removeWrappingDiv(container) : container,
      ).toMatchSnapshot();

      done();
    });
  });
};

export const isNull = (title, component) => {
  it(title, () => {
    const { container } = render(component);
    expect(container.firstChild).toBeNull();
  });
};

export const setWindowValue = (key, value) => {
  const windowValue = window[key];
  delete window[key];

  let newValue = value;

  if (value && typeof value === 'object') {
    newValue = {
      ...deepClone(windowValue),
      ...value,
    };
  }

  Object.defineProperty(window, key, {
    value: newValue,
    writable: true,
  });
};

export const resetWindowValue = (key, value) => {
  Object.defineProperty(window, key, {
    value,
    writable: true,
  });
};

export const suppressPropWarnings = warnings => {
  const { expectedWarnings } = window;
  if (expectedWarnings && Array.isArray(expectedWarnings)) {
    window.expectedWarnings = [...expectedWarnings, warnings];
  } else {
    window.expectedWarnings = [warnings];
  }
};

const errorIfMissingKey = (keys, object, message) => {
  keys.forEach(key => {
    if (!(key in object)) {
      throw new Error(`Missing value '${key}' in ${message}.`);
    }
  });
};

const checkKeysExistInBothObjects = (object1, object2, message1, message2) => {
  const object1Keys = Object.keys(object1);
  const object2Keys = Object.keys(object2);

  errorIfMissingKey(object1Keys, object2, message2);
  errorIfMissingKey(object2Keys, object1, message1);
};

const checkTypesOfExports = (
  actualExportsByName,
  actualExports,
  expectedExports,
  utilityName,
) => {
  actualExportsByName.forEach(actualExportName => {
    const actualExportValue = actualExports[utilityName][actualExportName];
    const expectedExport = expectedExports[utilityName][actualExportName];
    const typeCheck = typeof actualExportValue === expectedExport; // eslint-disable-line valid-typeof

    // if this fails it is likely that an export is missing from the unit test expectation
    expect(typeCheck).toBe(true);
  });
};

export const testUtilityPackages = (
  actualExports,
  expectedExports,
  packageName,
) => {
  const actualUtilities = Object.keys(actualExports);

  // check if the actual exported file has defined expected values to match and that the expected values are actually exported
  checkKeysExistInBothObjects(
    actualExports,
    expectedExports,
    `the actual utilities for '${packageName}'`,
    `the expected utilities for '${packageName}'`,
  );

  actualUtilities.forEach(utilityName => {
    const actualExportsByName = Object.keys(actualExports[utilityName]);

    // check if each of the actual exported consts have an expected value to match and that the expected values are actually exported
    checkKeysExistInBothObjects(
      actualExports[utilityName],
      expectedExports[utilityName],
      `the actual export for '${packageName}/${utilityName}'`,
      `the expected export for '${packageName}/${utilityName}'`,
    );

    checkTypesOfExports(
      actualExportsByName,
      actualExports,
      expectedExports,
      utilityName,
    );
  });
};
