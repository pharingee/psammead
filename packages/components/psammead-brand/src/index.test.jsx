import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { render } from '@testing-library/react';
import { C_POSTBOX, C_WHITE } from '@bbc/psammead-styles/colours';
import Brand from '.';

const svg = {
  group: (
    <g fillRule="evenodd">
      <path d="M84.32" />
    </g>
  ),
  viewbox: {
    height: 24,
    width: 167.95,
  },
  ratio: 6.9979,
};

describe('Brand', () => {
  shouldMatchSnapshot(
    'should render correctly with link provided',
    <Brand
      product="Default Brand Name"
      serviceLocalisedName="Service"
      svgHeight={24}
      maxWidth={280}
      minWidth={180}
      svg={svg}
      url="https://www.bbc.co.uk/news"
      backgroundColour={C_POSTBOX}
      logoColour={C_WHITE}
    />,
  );

  shouldMatchSnapshot(
    'should render correctly with link not provided',
    <Brand
      product="Default Brand Name"
      serviceLocalisedName="Service"
      svg={svg}
      svgHeight={24}
      maxWidth={280}
      minWidth={180}
      backgroundColour={C_POSTBOX}
      logoColour={C_WHITE}
    />,
  );

  shouldMatchSnapshot(
    'should render correctly with no service Localised Name',
    <Brand
      product="BBC News"
      svg={svg}
      svgHeight={24}
      maxWidth={280}
      minWidth={180}
      backgroundColour={C_POSTBOX}
      logoColour={C_WHITE}
    />,
  );

  shouldMatchSnapshot(
    'should render correctly with transparent borders',
    <Brand
      product="BBC News"
      svg={svg}
      svgHeight={24}
      maxWidth={280}
      minWidth={180}
      borderTop
      borderBottom
      backgroundColour={C_POSTBOX}
      logoColour={C_WHITE}
    />,
  );

  describe('assertions - visually hidden text', () => {
    it('should have role of text when serviceLocalisedName is provided', () => {
      const { container } = render(
        <Brand
          product="Default Brand Name"
          serviceLocalisedName="Service"
          svgHeight={24}
          maxWidth={280}
          minWidth={180}
          svg={svg}
          url="https://www.bbc.co.uk/news"
          backgroundColour={C_POSTBOX}
          logoColour={C_WHITE}
        />,
      );

      expect(container.querySelector('span').getAttribute('role')).toEqual(
        'text',
      );
    });

    it('should not have role of text when serviceLocalisedName is not provided', () => {
      const { container } = render(
        <Brand
          product="Default Brand Name"
          svgHeight={24}
          maxWidth={280}
          minWidth={180}
          svg={svg}
          url="https://www.bbc.co.uk/news"
          backgroundColour={C_POSTBOX}
          logoColour={C_WHITE}
        />,
      );

      expect(container.querySelector('span').getAttribute('role')).toBeNull();
    });
  });
});
