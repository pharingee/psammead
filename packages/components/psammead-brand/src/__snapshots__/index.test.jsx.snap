// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand should render correctly with link not provided 1`] = `
.c3 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c1 {
  max-width: 80rem;
  margin: 0 auto;
}

.c0 {
  background-color: #B80000;
  height: 3.5rem;
  width: 100%;
  padding: 0 0.5rem;
}

.c2 {
  box-sizing: content-box;
  color: #FFFFFF;
  fill: currentColor;
  padding-top: 1rem;
  padding-bottom: 0.75rem;
  height: 1.5rem;
  width: 100%;
  max-width: 17.5rem;
  min-width: 11.25rem;
}

.c4:hover .c2,
.c4:focus .c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 0.25rem solid #FFFFFF;
}

@media (min-width:25rem) {
  .c0 {
    height: 5rem;
    padding: 0 1rem;
  }
}

@media (min-width:25rem) {
  .c2 {
    padding-top: 1.75rem;
    padding-bottom: 1.5rem;
  }
}

@media screen and (-ms-high-contrast:active),print {
  .c2 {
    fill: windowText;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="c2"
      focusable="false"
      height="24"
      viewBox="0 0 167.95 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill-rule="evenodd"
      >
        <path
          d="M84.32"
        />
      </g>
    </svg>
    <span
      class="c3"
      role="text"
    >
      <span
        lang="en-GB"
      >
        Default Brand Name
      </span>
      , 
      Service
    </span>
  </div>
</div>
`;

exports[`Brand should render correctly with link provided 1`] = `
.c5 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c1 {
  max-width: 80rem;
  margin: 0 auto;
}

.c0 {
  background-color: #B80000;
  height: 3.5rem;
  width: 100%;
  padding: 0 0.5rem;
}

.c3 {
  display: inline-block;
  width: 100%;
  max-width: 17.5rem;
  min-width: 11.25rem;
}

.c4 {
  box-sizing: content-box;
  color: #FFFFFF;
  fill: currentColor;
  padding-top: 1rem;
  padding-bottom: 0.75rem;
  height: 1.5rem;
  width: 100%;
  max-width: 17.5rem;
  min-width: 11.25rem;
}

.c2:hover .c4,
.c2:focus .c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 0.25rem solid #FFFFFF;
}

@media (min-width:25rem) {
  .c0 {
    height: 5rem;
    padding: 0 1rem;
  }
}

@media (min-width:25rem) {
  .c4 {
    padding-top: 1.75rem;
    padding-bottom: 1.5rem;
  }
}

@media screen and (-ms-high-contrast:active),print {
  .c4 {
    fill: windowText;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <a
      class="c2 c3"
      href="https://www.bbc.co.uk/news"
    >
      <svg
        aria-hidden="true"
        class="c4"
        focusable="false"
        height="24"
        viewBox="0 0 167.95 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill-rule="evenodd"
        >
          <path
            d="M84.32"
          />
        </g>
      </svg>
      <span
        class="c5"
        role="text"
      >
        <span
          lang="en-GB"
        >
          Default Brand Name
        </span>
        , 
        Service
      </span>
    </a>
  </div>
</div>
`;

exports[`Brand should render correctly with no service Localised Name 1`] = `
.c3 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c1 {
  max-width: 80rem;
  margin: 0 auto;
}

.c0 {
  background-color: #B80000;
  height: 3.5rem;
  width: 100%;
  padding: 0 0.5rem;
}

.c2 {
  box-sizing: content-box;
  color: #FFFFFF;
  fill: currentColor;
  padding-top: 1rem;
  padding-bottom: 0.75rem;
  height: 1.5rem;
  width: 100%;
  max-width: 17.5rem;
  min-width: 11.25rem;
}

.c4:hover .c2,
.c4:focus .c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 0.25rem solid #FFFFFF;
}

@media (min-width:25rem) {
  .c0 {
    height: 5rem;
    padding: 0 1rem;
  }
}

@media (min-width:25rem) {
  .c2 {
    padding-top: 1.75rem;
    padding-bottom: 1.5rem;
  }
}

@media screen and (-ms-high-contrast:active),print {
  .c2 {
    fill: windowText;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="c2"
      focusable="false"
      height="24"
      viewBox="0 0 167.95 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill-rule="evenodd"
      >
        <path
          d="M84.32"
        />
      </g>
    </svg>
    <span
      class="c3"
    >
      BBC News
    </span>
  </div>
</div>
`;

exports[`Brand should render correctly with transparent borders 1`] = `
.c3 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c1 {
  max-width: 80rem;
  margin: 0 auto;
}

.c0 {
  background-color: #B80000;
  height: 3.5rem;
  width: 100%;
  padding: 0 0.5rem;
  border-top: 0.0625rem solid transparent;
  border-bottom: 0.0625rem solid transparent;
}

.c2 {
  box-sizing: content-box;
  color: #FFFFFF;
  fill: currentColor;
  padding-top: 1rem;
  padding-bottom: 0.75rem;
  height: 1.5rem;
  width: 100%;
  max-width: 17.5rem;
  min-width: 11.25rem;
}

.c4:hover .c2,
.c4:focus .c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 0.25rem solid #FFFFFF;
}

@media (min-width:25rem) {
  .c0 {
    height: 5rem;
    padding: 0 1rem;
  }
}

@media (min-width:25rem) {
  .c2 {
    padding-top: 1.75rem;
    padding-bottom: 1.5rem;
  }
}

@media screen and (-ms-high-contrast:active),print {
  .c2 {
    fill: windowText;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="c2"
      focusable="false"
      height="24"
      viewBox="0 0 167.95 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill-rule="evenodd"
      >
        <path
          d="M84.32"
        />
      </g>
    </svg>
    <span
      class="c3"
    >
      BBC News
    </span>
  </div>
</div>
`;
